### PR TITLE
separated tabs from ui crate

### PR DESCRIPTION
**Pull request by rewin123**
Currently its just code moving. Want to split separations PRs in multiple steps

### CODE REVIEW

**Comment by naomijub** [comment]
Should this target `main` or `v0.6`? Being a major change, I would target v0.6

**Comment by rewin123** [comment]
> Should this target `main` or `v0.6`? Being a major change, I would target v0.6

Oh, yes. Misselect

File: Cargo.toml [hunk 1]
```diff
@@ -14,17 +14,12 @@ repository = "https://github.com/rewin123/space_editor"
...
-    "crates/prefab", 
-    "crates/shared", 
-    "crates/undo", 
-    "crates/persistence",
-    "crates/editor_core", 
-    "crates/editor_ui", 
+    "crates/*",
     "modules/bevy_xpbd_plugin"
-]
+, "crates/editor_tabs"]
```
Comment (naomijub) [inline Cargo.toml:19]: this is already included in `"crates/*"`

File: Cargo.toml [hunk 2]
```diff
@@ -14,17 +14,12 @@ repository = "https://github.com/rewin123/space_editor"
...
 
 [workspace.package]
-version = "0.5.0"
+version = "0.6.0"
```
Comment (naomijub) [inline Cargo.toml:22]: 🥁

File: crates/editor_tabs/src/colors.rs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/// Colors used in editor
+use bevy_egui::egui::{Color32, Stroke};
```
Comment (naomijub) [inline crates/editor_tabs/src/colors.rs:2]: I think color, sizing and typography would live better in the UI crate, as they are ui elements. If this crate is the one responsible for the application ui, maybe call this editor_ui and the other editor_ui_extensions/editor_ui_components?

File: crates/editor_tabs/src/sizing.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
+use bevy::prelude::*;
+use bevy_inspector_egui::prelude::*;
+use egui_dock::egui::{Color32, RichText};
+
+#[derive(Resource, Clone, PartialEq, Reflect, InspectorOptions)]
+#[reflect(Resource, Default, InspectorOptions)]
+pub struct Sizing {
```
Comment (naomijub) [inline crates/editor_tabs/src/sizing.rs:7]: same as color comment

File: crates/editor_ui/src/change_chain.rs [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
+use space_editor_tabs::prelude::*;
 use space_undo::ChangeChain;
 
 pub struct ChangeChainViewPlugin;
 
 impl Plugin for ChangeChainViewPlugin {
     fn build(&self, app: &mut App) {
         app.editor_tab_by_trait(
-            super::editor_tab::EditorTabName::Other("Change Chain".to_string()),
+            EditorTabName::Other("Change Chain".to_string()),
```
Comment (naomijub) [inline crates/editor_ui/src/change_chain.rs:11]: UX suggested that `Change Chain` is a non intuitive name for Undo/Redo logic, Maybe we could call it `Undo Chain`, `Undo/Redo Config` or `Undo Max Chain`?

File: crates/editor_tabs/src/colors.rs (hunk 3 above)
Comment (rewin123) [reply crates/editor_tabs/src/colors.rs:2]: Yes, this is undoubtedly better located in editor_ui crate. However, these colours are referenced by editor_tabs and you need to get rid of that. However, the PR is already quite large, so I decided to stop with just shuffling the code for now. And the improvement of the separations will be in multiple PRs

File: crates/editor_tabs/src/sizing.rs (hunk 4 above)
Comment (rewin123) [reply crates/editor_tabs/src/sizing.rs:7]: Same as above

File: crates/editor_ui/src/change_chain.rs (hunk 5 above)
Comment (rewin123) [reply crates/editor_ui/src/change_chain.rs:11]: We can! But in another PR, this PR only about code moving

File: crates/editor_ui/Cargo.toml [hunk 6]
```diff
@@ -22,6 +22,7 @@ space_prefab.workspace = true
 space_undo.workspace = true
 space_persistence.workspace = true
 space_shared.workspace = true
+space_editor_tabs.workspace = true
```
[the remark this answers is not in the document]
Comment (rewin123) [reply crates/editor_ui/Cargo.toml:25]: No, it's in the plans

**Pull request merged**